spec/models/job_proxy_dispatcher_spec.rb

```diff
@@ -276,45 +271,139 @@
   end
 
   context "limiting number of smart state analysis running on one server" do
+    let(:job) { Job.create_job("VmScan", :agent_id => @server.id, :name => "Hello - 1") }
     before(:each) do
-      Job.create_job("VmScan", :agent_id => @server.id, :name => "Hello - 1")
-         .update_attributes(:dispatch_status => "any")
       Job.create_job("VmScan", :agent_id => @server.id, :name => "Hello - 2")
          .update_attributes(:dispatch_status => "active")
       Job.create_job("VmScan", :agent_id => @server.id, :name => "Hello - 3")
          .update_attributes(:dispatch_status => "active")
-      @proxy_dispatcher = JobProxyDispatcher.new
     end
 
     describe "#busy_proxies" do
       it "it returns hash with number of not finished jobs with dispatch status 'active' for each MiqServer" do
-        expect(@proxy_dispatcher.busy_proxies).to eq "MiqServer_#{@server.id}" => 2
+        expect(dispatcher.busy_proxies).to eq "MiqServer_#{@server.id}" => 2
       end
     end
 
     describe "#assign_proxy_to_job" do
       it "increses by 1 number of jobs (how busy server is) for server" do
-        job = Job.create_job("VmScan", :agent_id => @server.id, :name => "Hello - 4")
-        expect(@proxy_dispatcher.busy_proxies).to eq "MiqServer_#{@server.id}" => 2
+        expect(dispatcher.busy_proxies).to eq "MiqServer_#{@server.id}" => 2
+        allow(dispatcher).to receive(:embedded_scan_resource).and_return(nil)
+        dispatcher.assign_proxy_to_job(@server, job)
+        expect(dispatcher.busy_proxies).to eq "MiqServer_#{@server.id}" => 3
```

Comment:
Wow, this change is much better!  It wasn't clear what it was doing before... now it's obvious.